### ISSUE TITLE
linter: API for eslint-plugin-oxlint

### ISSUE THREAD

**Issue by Boshen** [oxc-project/oxc#7517]
the eslint plugin `eslint-plugin-oxlint` does not parse the `rules.rs`.
Instead it looks recursivly into the rules directory and getting all informations (category: correctness, style, .., plugin-name: folder)

I am already getting annoyed because of the 3 big parts:

- Documentation in oxlint
- Detection in `oxlintrc.json` (maybe even oxlint cli, did not test it)
- Remapping in `eslint-plugin-oxlint`

_Originally posted by @Sysix in https://github.com/oxc-project/oxc/discussions/7242#discussioncomment-11401060_

**Comment by Boshen** [comment]
Need a json API for all lint rules.

**Comment by Sysix** [comment]
Bonus point: Able to output too it with the used configuration. 
So I can also check in the integration tests for the rule names and not only for the count :)

**Comment by camc314** [comment]
What should this look like? a public rust function exported from the crate? of from the oxlint cli?
or a task like the website docs (iirc)?

i might be able to do this next week if we can get that info nailed down

**Comment by Boshen** [comment]
From the oxlint cli.

**Comment by camc314** [comment]
@Sysix what's the ideal shape?
would:
```ts
type Rule = {
    category: string;
    source: string; // unicorn/typescript/react ect.
    name: string;
}

type Response = Rule[];
```

be ok? or is that missing some info?

**Comment by Sysix** [comment]
Looks perfect @camc314 

the current source code type is:

```ts
export interface Rule {
  value: string; // name
  scope: string; // source
  category: string;
  error?: string; // when we could not parse the rule with our parser
}
```

**Comment by camc314** [comment]
perfect, i'll rename to use those words

**Comment by Sysix** [comment]
feel free to change those words :) these are from the past and I also not like the name `scope` :)

**Comment by camc314** [comment]
😅 too late

🙂 i don't mind what we call it.


maybe `pluginName` and `name` ?

i think scope kinda makes sense though
